src/backend/api/meals.js

```diff
@@ -2,14 +2,83 @@ const express = require("express");
 const router = express.Router();
 const knex = require("../database");
 
+// /api/meals - GET - Returns all meals
 router.get("/", async (request, response) => {
   try {
     // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    const meals = await knex("meal").select("*");
+    if (!meals) {
+      return response.status(404)
+    } else {
+      response.json(meals);
+    }
   } catch (error) {
     throw error;
   }
 });
 
+// /api/meals - POST - Adds a new meal to the database
+router.post("/", async (request, response) => {
+  try {
+    const meal = request.body;
+    await knex("meal").insert(meal);
+
+    if (!meal) {
+      return response.status(404)
+    } else {
+      response.json(meal);
+    }
+  } catch (error) {
+    return response.status(400).send(error?.sqlMessage);
+  }
+});
+
+// /api/meals/:id - GET - 	Returns the meal by id
+router.get("/:id", async (request, response) => {
+  try {
+    const id = request.params.id;
+    const meal = await knex("meal").select("*").where({ id: id });
+    if (!meal) {
+      return response.status(404)
+    } else {
+      response.json(meal);
+    }
+  } catch (error) {
+    throw error;
+  }
```

Comment:
This will cause an error on the server and not respond to the user. Could you think of a better way here?